src/sardana/taurus/qt/qtgui/extra_macroexecutor/macrobutton.py

```diff
@@ -147,6 +147,11 @@ def _doorStateChanged(self, state):
         if state not in [PyTango.DevState.ON, PyTango.DevState.ALARM] and not self.ui.button.isChecked():
             door_available = False
 
+        if state in [PyTango.DevState.RUNNING]:
+            self.setButtonText(self.macro_name + ": Click to abort")
```

Comment:
I may be missing something, but I do not see where is `setButtonText` defined...